### PR TITLE
Translate some tests with UDP functionality to TCP

### PR DESCRIPTION
**Pull request by guilhermelawless**
Test `network.last_contacted` was failing due to not setting `last_packet_received` in TCP channels, added this behavior though it's only used for tests at the moment.

Does not touch tests specific to UDP:
- network.send_insufficient_work_udp
- network.replace_port
- network.send_node_id_handshake
- node.connect_after_junk
- node_flags.disable_udp
- telemetry.remove_peer_different_genesis_udp

Some tests have UDP functionality that can also be removed in the future:
- network.duplicate_detection
- node.rep_remove

### CODE REVIEW

File: nano/core_test/node.cpp [hunk 1]
```diff
@@ -1583,10 +1569,15 @@ TEST (node, DISABLED_fork_stale)
 	nano::system system2 (1);
 	auto & node1 (*system1.nodes[0]);
 	auto & node2 (*system2.nodes[0]);
-	node2.bootstrap_initiator.bootstrap (node1.network.endpoint ());
-	std::shared_ptr<nano::transport::channel> channel (std::make_shared<nano::transport::channel_udp> (node2.network.udp_channels, node1.network.endpoint (), node2.network_params.protocol.protocol_version));
-	auto vote = std::make_shared<nano::vote> (nano::dev_genesis_key.pub, nano::dev_genesis_key.prv, 0, std::vector<nano::block_hash> ());
-	node2.rep_crawler.response (channel, vote);
+	node2.network.tcp_channels.start_tcp (node1.network.endpoint ());
```
Comment (SergiySW) [inline nano/core_test/node.cpp]: I think this replacement is not related to initial test design. It's disabled anyway, but still

File: nano/core_test/network.cpp [hunk 2]
```diff
@@ -136,58 +132,46 @@ TEST (network, send_node_id_handshake_tcp)
...
 	auto timestamp_before_keepalive = channel2->get_last_packet_received ();
+	auto keepalive_count = node0->stats.count (nano::stat::type::message, nano::stat::detail::keepalive, nano::stat::dir::in);
 	node1->network.send_keepalive (channel1);
-	ASSERT_TIMELY (10s, node0->stats.count (nano::stat::type::message, nano::stat::detail::keepalive, nano::stat::dir::in) >= 2);
+	ASSERT_TIMELY (3s, node0->stats.count (nano::stat::type::message, nano::stat::detail::keepalive, nano::stat::dir::in) > keepalive_count);
 	ASSERT_EQ (node0->network.size (), 1);
 	auto timestamp_after_keepalive = channel2->get_last_packet_received ();
 	ASSERT_GT (timestamp_after_keepalive, timestamp_before_keepalive);
-
-	node1->stop ();
```
Comment (SergiySW) [inline nano/core_test/network.cpp:164]: Not sure if removing `stop ()` function in this test & some below is safe enough

**Comment by zhyatt** [comment]
@SergiySW Are you able to make updates to this PR to get is ready for merging?

File: nano/core_test/network.cpp (hunk 2 above)
Comment (SergiySW) [reply nano/core_test/network.cpp:164]: `system.nodes.push_back (node1);` should solve this

**Review by SergiySW — APPROVED** [review]

**Review by wezrule — APPROVED** [review]
